### ISSUE TITLE
Allow arbitrary units for waverange grids

### ISSUE THREAD

**Issue by ritzvik** [radis/radis#54]
Opening new issue as directed in #53 @erwanp 

Units can be easily defined by using the `units` module of `astropy`
For example,
```
>>> from astropy import units as u
>>> l = 5*u.cm
>>> l.to(u.m)
<Quantity 0.05 m>
>>> type(l)
<class 'astropy.units.quantity.Quantity'>
>>> l.value
5.0
```

Can `astropy` be added as a dependency to generalize units throughout `radis` eventually ?

**Comment by ritzvik** [comment]
Initially, `radis/spectrum.py` can be augmented with this new feature, allowing user to give inputs using old ways or using `astropy.units` and the datatype can be checked by `isinstance()`.

Should I proceed with this?

@erwanp

**Comment by erwanp** [comment]
Hi @ritzvik ! Thanks for jumping on this Issue! 

Please go ahead! Here are a few hints & questions that may help: 

We currently have 3 kind of units: wavenumbers, wavelengths in vacuum, and wavelengths in air. For instance the `Spectrum.get()` method has:
```
        if wunit == 'cm-1':
            w = self.get_wavenumber(vartype, copy=copy)
        elif wunit == 'nm':
            w = self.get_wavelength(medium='air', which=vartype, copy=copy)
        elif wunit == 'nm_vac':
            w = self.get_wavelength(medium='vacuum', which=vartype, copy=copy)
```
- the default wavelength could be wavelength in air, and wavelengths in vacuum could be identified as ending with `_vac` (then trimmed and analyzed by Astropy). What do you think? 

- the `is_homogeneous` function in `radis.phys.units` can be used to differentiate between wavelength (~ `nm`) and wavenumbers (~ `cm-1`). The `conv2` function may also help!

- These functions are currently based on the [pint](https://github.com/hgrecco/pint) package and used to convert Intensity units. We could stick to it, but switching to `astropy.units` entirely sounds like a good idea, as we're already using the `astroquery` package from the `astropy` ecosystem.

- arbitrary inputs could also be given to the LBL calculations, in [calc_spectrum](https://radis.readthedocs.io/en/latest/source/radis.lbl.calc.html#radis.lbl.calc.calc_spectrum) or [SpectrumFactory](https://radis.readthedocs.io/en/latest/source/radis.lbl.factory.html#radis.lbl.factory.SpectrumFactory). That would be neat!

**Comment by erwanp** [comment]
Closed by #55
